presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -329,7 +332,11 @@ public ConnectorTableHandle getTableHandleForStatisticsCollection(ConnectorSessi
             }
         });
 
-        return handle;
+        HiveTableHandle table = handle;
+        return partitionValuesList
```

Comment:
Why is this needed now?

Reply:
The listing has to be done somewhere. Doing it here makes sense as we'd otherwise need to move this into `HivePartitionManager.getOrLoadPartitions()`, with the special case logic for the analyze partition values list.